lib/sharepoint/client.rb

```diff
@@ -641,5 +685,42 @@ def remove_double_slashes(str)
               .gsub('https:/', 'https://')
     end
 
+    def build_query_params(query)
+      query_params = []
+
+      query.each do |field, value|
+        query_params << "$#{field}=#{value}"
+      end
+
+      query_params.join('&')
+    end
+
+    def parse_lists_in_site_response(response_body)
+      json_response = JSON.parse(response_body)
+      results = json_response.dig('d', 'results')
+
+      results.map do |result|
+        OpenStruct.new(result.map { |k, v| [k.underscore.to_sym, v] }.to_h)
+      end
+    end
+
+    def update_object_metadata(metadata, new_metadata, site_path = '')
```

Comment:
`update_metadata` could be refactored to use this method, but it is not tested so I'm leaving this here to avoid any potential breaking change.